### PR TITLE
[doc only] 1568767: Document `sdk_generator.gradle` options

### PR DESCRIPTION
**Pull request by mdboom**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: docs/user/android-build-configuration-options.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## `gleanNamespace`
+
+Should be set to the Java/Kotlin namespace that the Glean API is in.  
+Normally, this should be left at the default, but it is useful to switch between different implementations of Glean.
+
+## `allowGleanInternal`
+
+This allows a library to define metrics and pings in namespaces that are reserved for the Glean library itself.
+This should never be set from an application or library other than Glean.
```
Comment (mdboom) [inline docs/user/android-build-configuration-options.md]: I'm not 100% sold we should document these last two: Telling users where our "backdoors" are might not be a good thing. But including them for completeness now and I'm happy to delete.

Comment (travis79) [reply docs/user/android-build-configuration-options.md]: I'm with you on this, I don't think documenting things that shouldn't be touched is a good idea.

File: docs/user/android-build-configuration-options.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ext.allowMetricsFromAAR = true
+```
+
+When this flag is set, every direct dependency of your library will be searched for a `metrics.yaml` file, and those metrics will be treated as the metrics as if they were defined by your library.
+That is, API wrappers accessible from your library will be generated for those metrics.
+
+The `metrics.yaml` can be added to the dependency itself by calling this on each relevant build variant:
+
+```groovy
+    variant.packageLibraryProvider.get().from("${topsrcdir}/path/metrics.yaml")
```
Comment (badboy) [inline docs/user/android-build-configuration-options.md]: No indentation needed

File: docs/user/android-build-configuration-options.md (hunk 1 above)
Comment (badboy) [reply docs/user/android-build-configuration-options.md]: Given this is user-facing, I agree. Let's leave it out.

Comment (Dexterp37) [reply docs/user/android-build-configuration-options.md]: I'd recommend moving them to our [internal docs](https://mozilla.github.io/glean/book/dev/core/internal/index.html), but I have no strong opinion on this :)

**Comment by badboy** [comment]
Rekicked tests, as Rust Beta was failing with out-of-memory. If that keeps happening, I'm inclined to un-require them for now, until we look into that.

File: docs/user/android-build-configuration-options.md (hunk 1 above)
Comment (mdboom) [reply docs/user/android-build-configuration-options.md]: Personally, I think it's fine that they are just documented in comments for now.

**Pull request merged**